src/layout/core/borders/container_draw.rs

```diff
@@ -33,21 +33,113 @@ impl ContainerDraw {
         x = gap / 2.0;
         w -= gap;
 
-        // Draw background of title bar
-        self.base.set_source_rgb(1.0, 0.0, 0.0);
-        self.base.set_color_source(title_color);
-        self.base.rectangle(x, 0.0, w, title_size);
-        self.base = try!(self.base.check_cairo());
-        self.base.fill();
-        self.base = try!(self.base.check_cairo());
-
-        // Draw title text
-        self.base.move_to(title_x, title_y);
-        self.base = try!(self.base.check_cairo());
-        self.base.set_color_source(title_font_color);
-        self.base = try!(self.base.check_cairo());
-        self.base.show_text(title.as_str());
-        self.base = try!(self.base.check_cairo());
+        // Children data necessary for tabbed and stacked containers
+        let child_data = self.base.inner().children.as_ref().map(
+            |children| (children.titles.len(), children.index)
+        );
+
+        // For some reason, when changing a container to be tabbed or stacked,
+        // the borders are drawn when the layout is set but the children are
+        // not, so the layout being Tabbed/Stacked doesn't guarantee having
+        // the children information.
+        match (self.base.inner().layout, child_data) {
+            (Some(Layout::Tabbed), Some((child_count, active_index))) => {
+
+                let tab_width = w / child_count as f64;
+
+                // Iterate over the indices to not borrow self
+                for i in 0..child_count {
+                    let active = i == active_index;
+
+                    let title = match self.base.inner().children.as_ref() {
```

Comment:
I couldn't find out how to do this without copying the String

Reply:
You can get around it by doing this:

```rust
                    self.base.show_text({
                        match self.base.inner().children.as_ref() {
                            Some(children) => &children.titles[i],
                            None => unreachable!()
                        }
                    });
```

Reply:
Nice, I tried something around that but didn't borrow check, but this does so i'm doing it